pkg/aggregator/sender.go

```diff
@@ -248,7 +254,7 @@ func (s *checkSender) sendMetricSample(
 		SampleRate:      1,
 		Timestamp:       timeNowNano(),
 		FlushFirstValue: flushFirstValue,
-		NoIndex:         noIndex,
+		NoIndex:         s.noIndex || noIndex,
```

Comment:
If we proceed with this path, the `noIndex` parameter will get removed and we'll only use the sender config. It's kept for no so that the existing stats telemetry continues to work as before.